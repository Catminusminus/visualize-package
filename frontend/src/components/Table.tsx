/**
 * Author of this code work, Yuki Mogi. c FFRI, Inc. 2019
 */

import * as React from 'react'
import { withStyles, WithStyles, createStyles } from '@material-ui/core/styles'
import Table from '@material-ui/core/Table'
import TableBody from '@material-ui/core/TableBody'
import TableCell from '@material-ui/core/TableCell'
import TableHead from '@material-ui/core/TableHead'
import TableRow from '@material-ui/core/TableRow'
import Paper from '@material-ui/core/Paper'
import { Stat } from '../modules'

const styles = theme =>
  createStyles({
    root: {
      width: '100%',
      marginTop: theme.spacing(3),
      overflowX: 'auto'
    },
    table: {
      minWidth: 700
    }
  })

const createData = ({
  packageName,
  lastCommit,
  latestPackage,
  maintainers
}: Stat) => {
  return {
    id: packageName,
    packageName,
    lastCommit,
    latestPackage,
    maintainers: maintainers.length
  }
}

interface Props extends WithStyles<typeof styles> {
  stats: Stat[]
}

const SimpleTable: React.FC<Props> = ({ classes, stats }) => {
  const rows = stats.map(stat => createData(stat))
  return (
    <Paper className={classes.root}>
      <Table className={classes.table}>
        <TableHead>
          <TableRow>
            <TableCell>Package</TableCell>
            <TableCell align="right">Last Commit</TableCell>
            <TableCell align="right">Latest Package</TableCell>
            <TableCell align="right">Maintainers</TableCell>
          </TableRow>
        </TableHead>
        <TableBody>
          {rows.map(row => (
            <TableRow key={row.id}>
              <TableCell component="th" scope="row">
                {row.packageName}
              </TableCell>
              <TableCell align="right">{row.lastCommit}</TableCell>
              <TableCell align="right">{row.latestPackage}</TableCell>
              <TableCell align="right">{row.maintainers}</TableCell>
            </TableRow>
          ))}
        </TableBody>
      </Table>
    </Paper>
  )
}

export default withStyles(styles)(SimpleTable)
