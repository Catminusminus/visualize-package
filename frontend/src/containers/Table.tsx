/**
 * Author of this code work, Yuki Mogi. c FFRI, Inc. 2019
 */

import * as React from 'react'
import TableComponent from '../components/Table'
import { useSelector } from 'react-redux'
import { State } from '../modules'

const statsSelector = (state: State) => state.reducer.stats

const Table = () => {
  const stats = useSelector(statsSelector)
  return <TableComponent stats={stats} />
}

export default Table
