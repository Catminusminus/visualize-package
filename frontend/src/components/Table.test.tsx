/**
 * Author of this code work, Yuki Mogi. c FFRI, Inc. 2019
 */

import * as React from 'react'
import renderer from 'react-test-renderer'
import Table from './Table'

it('Snapshot test of Table', () => {
  const result = renderer
    .create(
      <Table
        stats={[
          {
            packageName: '@ffri/sample1',
            lastCommit: '2018/12/31',
            latestPackage: '2018/12/01',
            maintainers: ['Adam', 'Bob']
          },
          {
            packageName: '@ffri/sample2',
            lastCommit: '2018/12/31',
            latestPackage: '2018/12/01',
            maintainers: ['Adam', 'Bob']
          }
        ]}
      />
    )
    .toJSON()
  expect(result).toMatchSnapshot()
})
