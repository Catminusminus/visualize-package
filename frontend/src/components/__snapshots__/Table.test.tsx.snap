// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test of Table 1`] = `
<div
  className="MuiPaper-root MuiPaper-elevation1 SimpleTable-root-1 MuiPaper-rounded"
>
  <table
    className="MuiTable-root SimpleTable-table-2"
  >
    <thead
      className="MuiTableHead-root"
    >
      <tr
        className="MuiTableRow-root MuiTableRow-head"
      >
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head"
          scope="col"
        >
          Package
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight"
          scope="col"
        >
          Last Commit
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight"
          scope="col"
        >
          Latest Package
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight"
          scope="col"
        >
          Maintainers
        </th>
      </tr>
    </thead>
    <tbody
      className="MuiTableBody-root"
    >
      <tr
        className="MuiTableRow-root"
      >
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body"
          scope="row"
        >
          @ffri/sample1
        </th>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
        >
          2018/12/31
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
        >
          2018/12/01
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
        >
          2
        </td>
      </tr>
      <tr
        className="MuiTableRow-root"
      >
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body"
          scope="row"
        >
          @ffri/sample2
        </th>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
        >
          2018/12/31
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
        >
          2018/12/01
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
        >
          2
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
